torchao/quantization/quant_primitives.py

```diff
@@ -374,57 +349,41 @@ def dynamically_quantize_per_channel(x, quant_min, quant_max, target_dtype):
     # assumes dense memory format
     # TODO(future): relax ^ as needed
 
-    # default setup for affine quantization of activations
-    eps = torch.finfo(torch.float32).eps
+    assert x.dim() == 2, "only support 2d Tensors"
 
-    # get min and max
-    min_val, max_val = torch.aminmax(x, dim=1)
-
-    # calculate scale and zero point based on min and max
-    # reference: https://fburl.com/code/srbiybme
-    min_val_neg = torch.min(min_val, torch.zeros_like(min_val))
-    max_val_pos = torch.max(max_val, torch.zeros_like(max_val))
-    device = min_val_neg.device
-
-    # reference: https://fburl.com/code/4wll53rk
-    max_val_pos = torch.max(-min_val_neg, max_val_pos)
-    scale = max_val_pos / (float(quant_max - quant_min) / 2)
-    # ensure scale is the same dtype as the original tensor
-    scale = torch.clamp(scale, min=eps).to(x.dtype)
-    zero_point = torch.zeros(min_val_neg.size(), dtype=torch.int64, device=device)
-
-    # quantize based on qmin/qmax/scale/zp
-    # reference: torch/ao/quantization/fx/_decomposed.py?lines=63
-    x_div = x.transpose(0, 1) / scale
-    x_round = torch.round(x_div)
-    x_zp = x_round + zero_point
-    x_zp = x_zp.transpose(0, 1)
-    quant = torch.clamp(x_zp, quant_min, quant_max).to(target_dtype)
+    eps = torch.finfo(torch.float32).eps
```

Comment:
Can it always be the float32 epsilon or should it be `x.dtype`?

Reply:
yeah we can do x.dtype

Reply:
You didn't do` x.dtype`, is that on purpose?